### PR TITLE
SO-4307: always use preferred synonym for reference set displays

### PR DESCRIPTION
**Pull request by cmark**

### CODE REVIEW

**Comment by cmark** [comment]
@kovacstomi Please execute the integration test suite and add some assertions to the existing test cases to verify the change.

**Review by apeteri — APPROVED** [review]
👍

**Comment by apeteri** [comment]
Converted PR to draft until the test case is added.

**Review by kovacstomi — COMMENTED** [review]
SNOMED API tests should run successfully.
Added ConceptMapMappingSearchRequest API test-cases to verify the changes.

File: snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/AllSnomedApiTests.java [hunk 1]
```diff
@@ -32,13 +32,24 @@
...
 import com.b2international.snowowl.snomed.core.request.ConceptSearchRequestSnomedTest;
 import com.b2international.snowowl.snomed.core.request.MemberSearchRequestSnomedTest;
+import com.b2international.snowowl.snomed.core.request.SnomedConceptMapSearchRequestTest;
 import com.b2international.snowowl.snomed.core.rest.branches.SnomedBranchingApiTest;
 import com.b2international.snowowl.snomed.core.rest.branches.SnomedMergeApiTest;
 import com.b2international.snowowl.snomed.core.rest.branches.SnomedMergeConflictTest;
 import com.b2international.snowowl.snomed.core.rest.branches.SnomedReviewApiTest;
 import com.b2international.snowowl.snomed.core.rest.classification.SnomedClassificationApiTest;
 import com.b2international.snowowl.snomed.core.rest.compare.SnomedMapTypeReferenceSetCompareTest;
-import com.b2international.snowowl.snomed.core.rest.components.*;
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/AllSnomedApiTests.java]: Restore this single import statement and configure your IDE to use `*` imports above `10` import statements.

File: snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/request/SnomedConceptMapSearchRequestTest.java [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+	
+	private void createSimpleMapTypeRefSetMember(final String rfId, final String sourceCode, final String targetCode) {
+		SnomedRequests.prepareNewMember()
+			.setId(UUID.randomUUID().toString())
+			.setReferenceSetId(rfId)
+			.setReferencedComponentId(sourceCode)
+			.setActive(true)
+			.setModuleId(Concepts.MODULE_SCT_CORE)
+			.setProperties(ImmutableMap.of(SnomedRf2Headers.FIELD_MAP_TARGET, targetCode))
+			.build(CODESYSTEM, "info@b2international.com", "New Reference Set")
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/request/SnomedConceptMapSearchRequestTest.java]: Use test user configuration from `RestExtensions.USER`, never hardcode a user name into any test or function.

**Review by cmark — COMMENTED** [review]
My review comments, unfortunately, I cannot approve/reject your changes @kovacstomi since I'm the creator of the PR.

**Comment by kovacstomi** [comment]
Fixed the requested changes @cmark

**Comment by apeteri** [comment]
Still 👍

**Pull request merged**